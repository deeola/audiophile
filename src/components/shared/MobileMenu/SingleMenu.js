import React from 'react'
import { TetiaryButton } from '../../Buttons/Button';
import { MenuImage, SingleMenuContainer, Text } from './MobileMenuStyling';


const SingleMenu = ({ labels="Shop", label,  alt, span=">", img, link, mobile, datatestid}) => {
  return (
    <SingleMenuContainer data-testid='singlemenu' mobile={mobile}>
        <MenuImage  data-testid='alt' src={img} alt={alt} />
        <Text data-testid='textlabel'>{label}</Text>
        <TetiaryButton datatestid={datatestid} label={labels} link={link} span={span} />
    </SingleMenuContainer>
  )
}

export default SingleMenu;



