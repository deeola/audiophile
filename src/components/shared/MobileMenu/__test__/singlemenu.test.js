import { render, cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import SingleMenu from "../SingleMenu";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders single menu correctly", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <SingleMenu />
    </BrowserRouter>,
    div
  );
});

it("renders shop test correctly and have contain an a tag", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <SingleMenu alt="alt" label="Shop" />
    </BrowserRouter>
  );
  expect(getByTestId("textlabel")).toHaveTextContent("Shop");
  expect(getByTestId("singlemenu")).toContainHTML("a");
  expect(getByTestId("alt")).toHaveAttribute("alt");
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <SingleMenu></SingleMenu>
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
