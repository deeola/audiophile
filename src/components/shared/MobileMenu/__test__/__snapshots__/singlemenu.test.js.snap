// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-dkzDqf kkLfEl"
  data-testid="singlemenu"
>
  <img
    className="sc-eCYdqJ jebgBi"
    data-testid="alt"
  />
  <p
    className="sc-hKMtZM"
    data-testid="textlabel"
  />
  <a
    className="sc-bczRLJ hcrMJw"
    color="tertiary"
    hover="tertiary"
    href=""
    onClick={[Function]}
  >
    Shop
    <span>
      &gt;
    </span>
  </a>
</div>
`;
